From: David Steele <dsteele@gmail.com>
Date: Wed, 3 Jun 2015 19:05:38 -0400
Subject: cpoauth.py: Proper error if no GNOME desktop

---
 cloudprint/cpoauth.py | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/cloudprint/cpoauth.py b/cloudprint/cpoauth.py
index c3992f0..7178ff8 100755
--- a/cloudprint/cpoauth.py
+++ b/cloudprint/cpoauth.py
@@ -163,9 +163,13 @@ def main():
         login += '@gmail.com'
 
     oauth = CPOAuth()
-    access, refresh = oauth.generate_tokens(login)
+    try:
+        access, refresh = oauth.generate_tokens(login)
 
-    print("%s\n%s\n%s" % (login, access, refresh))
+        print("%s\n%s\n%s" % (login, access, refresh))
+    except NameError:
+        print("GNOME environment required")
+        sys.exit(-1)
 
 if __name__ == '__main__':
     main()
